From 25dc1c7cde85a319ec3cdf4b634683f2e5b5f7ca Mon Sep 17 00:00:00 2001
From: "Wang, ArvinX" <arvinx.wang@intel.com>
Date: Thu, 6 Dec 2018 18:16:18 +0800
Subject: [PATCH] Fix "Guest" user option seen when logged in as "New User"

if the foreground user is the device owner, add the guest
user record.

Change-Id: Ie636ff43c14b6b62ac27caa3fe0500d714da5d88
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-70634
Signed-off-by: Wang, ArvinX <arvinx.wang@intel.com>
---
 .../src/com/android/systemui/statusbar/car/UserGridRecyclerView.java  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java b/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java
index 35575d2..1fdd4b7 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/car/UserGridRecyclerView.java
@@ -109,8 +109,8 @@ public class UserGridRecyclerView extends PagedListView implements
             userRecords.add(record);
         }
 
-        // Add guest user record if the foreground user is not a guest
-        if (!mUserManagerHelper.foregroundUserIsGuestUser()) {
+        // Add guest user record if the foreground user is the device owner
+        if (mUserManagerHelper.getForegroundUserInfo().isAdmin()) {
             userRecords.add(addGuestUserRecord());
         }
 
-- 
1.9.1

